packages/typst/build.sh

```diff
@@ -0,0 +1,21 @@
+TERMUX_PKG_HOMEPAGE=https://typst.app/
+TERMUX_PKG_DESCRIPTION="A new markup-based typesetting system that is powerful and easy to learn"
+TERMUX_PKG_LICENSE="Apache-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.1.0
+TERMUX_PKG_SRCURL=https://github.com/typst/typst/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=9da7543266c7776aed56c36b9fab95fdb7241674b426cfba287a21f4cf07c172
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_AUTO_UPDATE=true
+
+termux_step_make() {
+	termux_setup_rust
+
+	export TYPST_HASH=$(git ls-remote https://github.com/typst/typst refs/tags/v$TERMUX_PKG_VERSION | head -c 8)
```

Comment:
Using `git+` URL is allowed if packaging requires metadata from Git repository. Then the patch is not needed.